File: ruby/zork/v1/player.rb
class Player
  attr_accessor :name, :level, :current_room

  def initialize
    print "What is your name? "
    @name = gets.chomp
    @level = 1
  end

  def status
    puts "You are #@name, a Level #@level adventurer."
    puts "You are currently in #{@current_room.name}."
    puts @current_room.description if @current_room.description
  end
end

File: lua/zork/player.lua
local Room = require "room"

Player = {}
function Player:new(name, level, current_room)
   return Class.new(self, {
      name = name,
      level = level or 1,
      current_room = current_room
   })
end

function Player:status()
   print("You are " .. self.name .. ", a Level " .. self.level .. " adventurer.")
   print("You are currently in " .. self.current_room.name .. ".")
   if self.current_room.description then
      print(self.current_room.description)
   end
end

return Player
